### ISSUE TITLE
[PAID] [$250] Search - Saved search area is partially overlapped by url field

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#50592]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.47
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:** N/A
**Issue reported by:** Applause - Internal Team

## Action Performed:
1. Navigate to staging.new.expensify.com
2. Open Search tab
3. Open any saved search
4. Click 3-dot menu next to it > Rename
5. Enter an emoji or name with X:Y format and save it
6. Tap "v" at the top of a Search page to open the menu
7. Observe the saved search area.

## Expected Result:
Saved search area is fully visible

## Actual Result:
Saved search area is partially covered by the url field when opened after renaming it or right after saving the search

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [x] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

<details>
  <summary>Add any screenshot/video evidence</summary>

  
</details> 

https://github.com/user-attachments/assets/de3d2df7-2069-472d-a1f8-16dc837ae62a

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021846258880159780544</li>
        <li>Upwork Job ID: 1846258880159780544</li>
        <li>Last Price Increase: 2024-10-15</li>
<li>Automatic offers: </li>
<ul>
<li>ikevin127 | Reviewer | 104486937</li>
<li>mkzie2 | Contributor | 104486939</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @strepanier03</details>

**Comment by lanitochka17** [comment]
@strepanier03  FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors

**Comment by mkzie2** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Saved search area is partially covered by the url field when opened after renaming it or right after saving the search

### What is the root cause of that problem?
The issue happens if we exceed the `maxHeight` of the `ScrollView` in `PopoverMenu`. The wrap component of the `PopoverMenu` modal has `paddingVertical` is `16px` so the item inside `ScrollView` here can be overlapped 

https://github.com/Expensify/App/blob/39393b794b983e3164b213052a922aec69689c9b/src/components/PopoverMenu.tsx#L268

https://github.com/Expensify/App/blob/39393b794b983e3164b213052a922aec69689c9b/src/pages/Search/SearchTypeMenuNarrow.tsx#L193-L197

### What changes do you think we should make in order to solve the problem?

For this case, we should move the padding-bottom from the wrap component of `PopoverMenu` to the `ScrollView`.

https://github.com/Expensify/App/blob/39393b794b983e3164b213052a922aec69689c9b/src/pages/Search/SearchTypeMenuNarrow.tsx#L193-L197

To do that we need some steps like this

1. Introduce two new props in `PopoverMenu`: `innerContainerStyle` is the `innerContainerStyle` prop we will pass to the `PopoverWithMeasuredContent` in `PopoverMenu`, `scrollViewContainerStyle` is the container style we will apply to the `ScrollView` here

https://github.com/Expensify/App/blob/39393b794b983e3164b213052a922aec69689c9b/src/components/PopoverMenu.tsx#L268
 
 2. We will pass `innerContainerStyle` as `styles.pb0` and `scrollViewContainerStyle` as `styles.pb4` to `PopoverMenu` here
 
 https://github.com/Expensify/App/blob/39393b794b983e3164b213052a922aec69689c9b/src/pages/Search/SearchTypeMenuNarrow.tsx#L193
### What alternative solutions did you explore? (Optional)
NA

**Comment by strepanier03** [comment]
Working on this now.

**Comment by strepanier03** [comment]
I don't have an iOS device to repro this on, but I think this is a very minor visual issue. 

<img width="849" alt="image" src="https://github.com/user-attachments/assets/bf3b41b4-4362-4f62-bc78-42bd7aefe5a9">

**Comment by ikevin127** [comment]
♻️ Reviewing today...

**Comment by ikevin127** [comment]
@strepanier03 I tried reproducing this on iOS Simulator and had no luck.

##

I reviewed a PR recently (https://github.com/Expensify/App/pull/50244) which fixed the scrolling of the list on large layout devices where it wasn't working, and I think it's expected that when opening the popover menu, the list scroll is at the top when the layout has scrolling space expanded due to one or multiple saved search items, depending on device height.

Therefore I propose to close this as a non-issue. Otherwise I'd need more information on the device this was reproduced on and more detailed steps on how exactly - if I can reproduce then I'll also be able to test whether the existing proposal fixes the issue.

**Comment by mkzie2** [comment]
@ikevin127 You can try to save many saved searches and open the first saved search. Besides, I think the only problem here is that we have a space at the bottom.

<img width="384" alt="Screenshot 2024-10-16 at 10 51 33" src="https://github.com/user-attachments/assets/a959d6ea-4cf1-46f2-9e1e-c468ea662be1">

<img width="383" alt="Screenshot 2024-10-16 at 10 52 52" src="https://github.com/user-attachments/assets/68b43afc-0b0d-4619-bbeb-27969d87c57a">

**Comment by ikevin127** [comment]
Thanks for the info! I noticed the space at the bottom on my side as well even though it was not going over any of the saved search options.

> Which of our officially supported platforms is this issue occurring on?

I think there might be a mistake regarding the platforms on which the issue is occuring as the `✅ iOS: Native` is checked but all videos where this was reproduced are on `✅ iOS: mWeb Safari` from what I can see.

Before removing the white space from the bottom as suggested by the proposal, I want to ask whether the white space at the bottom was put there on purpose. If nobody knows I guess we can move forward with removing it.

This is how things look on my side (dark theme) and with Safari tab bar both on the bottom and on the top:


https://github.com/user-attachments/assets/284ce84c-0528-458f-b59b-e6ebf6f0c8cd

cc @luacmartins @Kicu @Expensify/design Tagging all which either worked on `SearchFiltersNarrow.tsx` or might want have a say in whether we want that bottom space removed or anything else changed when it comes to this issue's **Expected result**.

> [!note]
> ### My take
> I think that the bottom spacing referenced in https://github.com/Expensify/App/issues/50592#issuecomment-2415674664 is there for a reason as it matches the popover's side / top spacing (see below) and back to this issue: when we have multiple saved search items in the list or even if there's just one and the height of the device is small -> it's expected we would have overlapping while still being able to scroll -> therefore not an issue.
>
> It would only be an issue if the overlapping would not allow scrolling and the option item would be cut by that spacing without being able to scroll to see it fully.
>
> But maybe some of you might see this from a different perspective, hence I tagged you and opened this up for discussion.

| Top / Bottom Spacing | Side Spacing |
| ------------- | ------------- |
| <img width="280" alt="Screenshot 2024-10-16 at 14 44 24" src="https://github.com/user-attachments/assets/dc8bbd1f-1af4-40ee-a49c-76a545e9c107"> | <img width="280" alt="Screenshot 2024-10-16 at 14 44 10" src="https://github.com/user-attachments/assets/e4cb9aa3-cf6b-4c3e-8a94-0e3115ef7294"> |

**Comment by shawnborton** [comment]
I think the bottom space is fine to be there, but it should scroll with the entire content of the modal. So basically we want these kinds of modals to have 16px padding on the top and bottom, but that padding should be considered as part of the scrollable area that is kept within the overall modal height, if that makes sense.

**Comment by mkzie2** [comment]
@shawnborton I think we should only move the padding bottom to the scrollable area. Moving the padding-top to the scrollable area the content will be close to the top edge when scrolling.

**Comment by shawnborton** [comment]
I don't think I agree with you, I think both top and bottom should be in the scrollable area. But let's see some videos to compare :)

**Comment by mkzie2** [comment]
@shawnborton Here is the video

- Only the bottom padding is in the scrollable area

https://github.com/user-attachments/assets/2175da75-e09f-4f1d-80bb-112d2000cf61


- Both top and bottom padding are in the scrollable area

https://github.com/user-attachments/assets/dcb37670-bf67-49b9-95fb-e8eab7571eae

**Comment by shawnborton** [comment]
I think the second video (both top and bottom can scroll) looks much better to me. Curious what the rest of the @Expensify/design thinks though!

**Comment by dannymcclain** [comment]
I agree—I think that feels more natural/expected to me.

**Comment by mkzie2** [comment]
Updated [proposal](https://github.com/Expensify/App/issues/50592#issuecomment-2405694436).

@ikevin127 The design is confirmed. Please help to take a look at my proposal again.

**Comment by dubielzyk-expensify** [comment]
100% with the second video 👍

**Comment by ikevin127** [comment]
cc @luacmartins for assignment here based on proposal review from https://github.com/Expensify/App/issues/50592#issuecomment-2420474775.

**Comment by ikevin127** [comment]
@mkzie2 Please let us know when we can expect a PR to be ready for review 🧑‍💻

**Comment by ikevin127** [comment]
* [x]  [@ikevin127] The PR that introduced the bug has been identified. Link to the PR: https://github.com/Expensify/App/pull/40419.
* [x]  [@ikevin127] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: https://github.com/Expensify/App/pull/40419/files#r1819876695.
* [x]  [@ikevin127] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion: **N/A**.
* [x]  [@ikevin127] Determine if we should create a regression test for this bug.
* [x]  [@ikevin127] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.

### Regression Test Proposal

1. Go to Search.
2. Create several saved search items if none (to make the popover scrollable).
3. Press the header's dropdown button.
4. Verify there's top padding in the popover.
5. Scroll the list.
6. Verify there's NO top AND bottom padding while scrolling.
7. Scroll to the bottom.
8. Verify there's bottom padding.

   **In other words, the padding is scrolled with the list instead of being fixed at the top or bottom.**

Do we agree 👍 or 👎.

**Comment by strepanier03** [comment]
Reg test submitted. Handling payments now.

**Comment by strepanier03** [comment]
### Payment Summary

- [x] $250 - @ikevin127 via [Upwork](https://www.upwork.com/nx/wm/offer/104486937) (Reviewer)
- [x] $250 - @mkzie2 via [Upwork](https://www.upwork.com/nx/wm/offer/104486939) (Contributor)
